### DISCUSSION TITLE
New Arch Support

### DISCUSSION THREAD

**Discussion by manage-some** [tr3v3r/react-native-esc-pos-printer#170] [Ideas]
We've release a new update on playstore that uses New Arch of React Native. However on old android devices, we have been reported that the app breaks. Upon investigation, we found out that it's because of this library specially in the case when app is connected to 3-4 printers and queue stacks up. When we disable New Arch, it seems to work fine on old android devices as well.

**Comment by tr3v3r** [comment]
Hello @manage-some 

Thanks for the report.

Let's clarify some important points first:

1) What lib version and RN version are you using?
2) Is it a bare or expo project?
3) Did you manage to reproduce the issue in DEV env?
4) Do you have any crash reports so we can have details?
5) Could you provide the way you're doing printer jobs?
6) How are you using it with new architecture? I mean I haven't done anything to support it yet...

Thanks

**Reply by manage-some** [reply]
1 - "react-native": "0.74.4"
2 - It is a bare RN project
3 - No, We're unable to reproduce it with the android devices we have. We can't use android studio because printer sdk is not supported on virtual android devices so we can only test with limited number of physical devices we have. However the issue was reported by 3 clients who were using old android devices. So we had to release a new update and disabled new arch which fixed the issue.
4-  Yes, we do. I'll attach here
5 - Print Jobs work fine, the error only occurs when we turn off the wifi or turn off the printer and on retry the app crashes
6 - We haven't done any changes, we just enabled new arch and it seemed to work with the android devices we have but once we release it to prod then some clients report the issue and then we checked the logs

**Reply by manage-some** [reply]
`*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
pid: 0, tid: 19204 >>> com.pos <<<

backtrace:
  #00  pc 0x000000000004e40c  /apex/com.android.runtime/lib64/bionic/libc.so (abort+164)
  #01  pc 0x0000000000014074  /data/app/~~gnRyG4hsoAX22VNJ6xj2AA==/com.pos-zcKgxdZkez4VUzq_MZjKUg==/split_config.arm64_v8a.apk!libglog.so (google::logging_fail()+14385152) (BuildId: c92b9cad78435c4b)
  #02  pc 0x0000000000012d68  /data/app/~~gnRyG4hsoAX22VNJ6xj2AA==/com.pos-zcKgxdZkez4VUzq_MZjKUg==/split_config.arm64_v8a.apk!libglog.so (google::LogMessage::SendToLog()+920) (BuildId: c92b9cad78435c4b)
  #03  pc 0x00000000000133d4  /data/app/~~gnRyG4hsoAX22VNJ6xj2AA==/com.pos-zcKgxdZkez4VUzq_MZjKUg==/split_config.arm64_v8a.apk!libglog.so (google::LogMessage::Flush()+220) (BuildId: c92b9cad78435c4b)
  #04  pc 0x0000000000017aec  /data/app/~~gnRyG4hsoAX22VNJ6xj2AA==/com.pos-zcKgxdZkez4VUzq_MZjKUg==/split_config.arm64_v8a.apk!libglog.so (google::LogMessageFatal::~LogMessageFatal()+8) (BuildId: c92b9cad78435c4b)
  #05  pc 0x000000000003b4f0  /data/app/~~gnRyG4hsoAX22VNJ6xj2AA==/com.pos-zcKgxdZkez4VUzq_MZjKUg==/split_config.arm64_v8a.apk!libreact_nativemodule_core.so (std::__ndk1::__function::__func<facebook::react::(anonymous namespace)::createJavaCallback(facebook::jsi::Runtime&, facebook::jsi::Function&&, std::__ndk1::shared_ptr<facebook::react::CallInvoker>)::$_0, std::__ndk1::allocator<facebook::react::(anonymous namespace)::createJavaCallback(facebook::jsi::Runtime&, facebook::jsi::Function&&, std::__ndk1::shared_ptr<facebook::react::CallInvoker>)::$_0>, void (folly::dynamic)>::operator()(folly::dynamic&&)+21348352) (BuildId: a20e60bf4634ffc5)
  #06  pc 0x00000000000a6b38  /data/app/~~gnRyG4hsoAX22VNJ6xj2AA==/com.pos-zcKgxdZkez4VUzq_MZjKUg==/split_config.arm64_v8a.apk!libreactnativejni.so (facebook::jni::detail::MethodWrapper<void (facebook::react::JCxxCallbackImpl::*)(facebook::react::NativeArray*), &(facebook::react::JCxxCallbackImpl::invoke(facebook::react::NativeArray*)), facebook::react::JCxxCallbackImpl, void, facebook::react::NativeArray*>::dispatch(facebook::jni::alias_ref<facebook::jni::detail::JTypeFor<facebook::jni::HybridClass<facebook::react::JCxxCallbackImpl, facebook::react::JCallback>::JavaPart, facebook::react::JCallback, void>::_javaobject*>, facebook::react::NativeArray*&&)+100) (BuildId: 414dcce2b8e85959)
  #07  pc 0x00000000000a6a5c  /data/app/~~gnRyG4hsoAX22VNJ6xj2AA==/com.pos-zcKgxdZkez4VUzq_MZjKUg==/split_config.arm64_v8a.apk!libreactnativejni.so (facebook::jni::detail::FunctionWrapper<void (*)(facebook::jni::alias_ref<facebook::jni::detail::JTypeFor<facebook::jni::HybridClass<facebook::react::JCxxCallbackImpl, facebook::react::JCallback>::JavaPart, facebook::react::JCallback, void>::_javaobject*>, facebook::react::NativeArray*&&), facebook::jni::detail::JTypeFor<facebook::jni::HybridClass<facebook::react::JCxxCallbackImpl, facebook::react::JCallback>::JavaPart, facebook::react::JCallback, void>::_javaobject*, void, facebook::react::NativeArray*>::call(_JNIEnv*, _jobject*, facebook::jni::detail::JTypeFor<facebook::jni::HybridClass<facebook::react::NativeArray, facebook::jni::detail::BaseHybridClass>::JavaPart, facebook::jni::JObject, void>::_javaobject*, void (*)(facebook::jni::alias_ref<facebook::jni::detail::JTypeFor<facebook::jni::HybridClass<facebook::react::JCxxCallbackImpl, facebook::react::JCallback>::JavaPart, facebook::react::JCallback, void>::_javaobject*>, facebook::react::NativeArray*&&))+80) (BuildId: 414dcce2b8e85959)
  #08  pc 0x00000000000505d8  /data/app/~~gnRyG4hsoAX22VNJ6xj2AA==/com.pos-zcKgxdZkez4VUzq_MZjKUg==/oat/arm64/base.odex (art_jni_trampoline+152)
  #09  pc 0x0000000000133564  /apex/com.android.art/lib64/libart.so (art_quick_invoke_stub+548)
  #10  pc 0x00000000001a8a78  /apex/com.android.art/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+200)
  #11  pc 0x000000000031830c  /apex/com.android.art/lib64/libart.so (art::interpreter::ArtInterpreterToCompiledCodeBridge(art::Thread*, art::ArtMethod*, art::ShadowFrame*, unsigned short, art::JValue*)+376)
  #12  pc 0x000000000030e638  /apex/com.android.art/lib64/libart.so (bool art::interpreter::DoCall<false, false>(art::ArtMethod*, art::Thread*, art::ShadowFrame&, art::Instruction const*, unsigned short, art::JValue*)+996)
  #13  pc 0x000000000067eb4c  /apex/com.android.art/lib64/libart.so (MterpInvokeDirect+576)
  #14  pc 0x000000000012d914  /apex/com.android.art/lib64/libart.so (mterp_op_invoke_direct+20)
  #15  pc 0x00000000004903fa  /data/app/~~gnRyG4hsoAX22VNJ6xj2AA==/com.pos-zcKgxdZkez4VUzq_MZjKUg==/oat/arm64/base.vdex (com.facebook.react.bridge.CxxCallbackImpl.invoke+8)
  #16  pc 0x000000000067e1a8  /apex/com.android.art/lib64/libart.so (MterpInvokeInterface+1808)
  #17  pc 0x000000000012da14  /apex/com.android.art/lib64/libart.so (mterp_op_invoke_interface+20)
  #18  pc 0x0000000000977820  /data/app/~~gnRyG4hsoAX22VNJ6xj2AA==/com.pos-zcKgxdZkez4VUzq_MZjKUg==/oat/arm64/base.vdex (com.zmxv.RNSound.RNSoundModule$2.onError+72)
  #19  pc 0x0000000000305c34  /apex/com.android.art/lib64/libart.so (art::interpreter::Execute(art::Thread*, art::CodeItemDataAccessor const&, art::ShadowFrame&, art::JValue, bool, bool) (.llvm.6432830707213048161)+268)
  #20  pc 0x000000000066b1d8  /apex/com.android.art/lib64/libart.so (artQuickToInterpreterBridge+780)
  #21  pc 0x000000000013cff8  /apex/com.android.art/lib64/libart.so (art_quick_to_interpreter_bridge+88)
  #22  pc 0x000000000069d148  /system/framework/arm64/boot-framework.oat (android.media.MediaPlayer$EventHandler.handleMessage+2552)
  #23  pc 0x00000000006348b4  /system/framework/arm64/boot-framework.oat (android.os.Handler.dispatchMessage+180)
  #24  pc 0x0000000000637dac  /system/framework/arm64/boot-framework.oat (android.os.Looper.loop+1516)
  #25  pc 0x00000000001337e8  /apex/com.android.art/lib64/libart.so (art_quick_invoke_static_stub+568)
  #26  pc 0x00000000001a8a94  /apex/com.android.art/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+228)
  #27  pc 0x000000000031830c  /apex/com.android.art/lib64/libart.so (art::interpreter::ArtInterpreterToCompiledCodeBridge(art::Thread*, art::ArtMethod*, art::ShadowFrame*, unsigned short, art::JValue*)+376)
  #28  pc 0x000000000030e638  /apex/com.android.art/lib64/libart.so (bool art::interpreter::DoCall<false, false>(art::ArtMethod*, art::Thread*, art::ShadowFrame&, art::Instruction const*, unsigned short, art::JValue*)+996)
  #29  pc 0x000000000067f428  /apex/com.android.art/lib64/libart.so (MterpInvokeStatic+548)
  #30  pc 0x000000000012d994  /apex/com.android.art/lib64/libart.so (mterp_op_invoke_static+20)
  #31  pc 0x0000000000495bd2  /data/app/~~gnRyG4hsoAX22VNJ6xj2AA==/com.pos-zcKgxdZkez4VUzq_MZjKUg==/oat/arm64/base.vdex (com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run+74)
  #32  pc 0x0000000000305c34  /apex/com.android.art/lib64/libart.so (art::interpreter::Execute(art::Thread*, art::CodeItemDataAccessor const&, art::ShadowFrame&, art::JValue, bool, bool) (.llvm.6432830707213048161)+268)
  #33  pc 0x000000000066b1d8  /apex/com.android.art/lib64/libart.so (artQuickToInterpreterBridge+780)
  #34  pc 0x000000000013cff8  /apex/com.android.art/lib64/libart.so (art_quick_to_interpreter_bridge+88)
  #35  pc 0x000000000015ab38  /apex/com.android.art/javalib/arm64/boot.oat (java.lang.Thread.run+72)
  #36  pc 0x0000000000133564  /apex/com.android.art/lib64/libart.so (art_quick_invoke_stub+548)
  #37  pc 0x00000000001a8a78  /apex/com.android.art/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+200)
  #38  pc 0x0000000000554c48  /apex/com.android.art/lib64/libart.so (art::JValue art::InvokeVirtualOrInterfaceWithJValues<art::ArtMethod*>(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, art::ArtMethod*, jvalue const*)+460)
  #39  pc 0x00000000005a3fe4  /apex/com.android.art/lib64/libart.so (art::Thread::CreateCallback(void*)+1308)
  #40  pc 0x00000000000b008c  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+64)
  #41  pc 0x00000000000503c8  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64)`

**Reply by tr3v3r** [reply]
@manage-some thanks for details. 
 
Please also mention library version :D

**Reply by manage-some** [reply]
This is library version

`"react-native-esc-pos-printer": "^4.1.3"`

**Comment by tr3v3r** [comment]
Ok, I see. It might be connected with the wrong way of allocating new threads on Andorid - I'll try to update it at my earliest convenience

**Reply by manage-some** [reply]
Thanks
